openapi/node.go

```diff
@@ -80,6 +99,7 @@ func Traverse(node Traversable, f TraversalFunc) (Traversable, error) {
 				continue
 			}
 			// Update Child Node
+			// TODO should child node pass down modified base path for easy resolution?
```

Comment:
basePath is only used in a subset of traversals. We should keep TraversalFunc's as generic as possible. I'd say no, it should not be passed down here.